hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/BlockStreamTransactionTranslator.java

```diff
@@ -51,8 +48,22 @@ public SingleTransactionRecord translate(
         Objects.requireNonNull(transaction, "transaction must not be null");
         Objects.requireNonNull(stateChanges, "stateChanges must not be null");
 
-        final var recordBuilder = TransactionRecord.newBuilder();
-        final var receiptBuilder = TransactionReceipt.newBuilder();
+        final var txnType = transaction.txn().bodyOrThrow().data().kind();
+        final var singleTxnRecord =
+                switch (txnType) {
+                    case UTIL_PRNG -> new UtilPrngTranslator().translate(transaction, stateChanges);
```

Comment:
I'd like to refactor `new UtilPrngTranslator()` to a factory method at some point, for testability. Doesn't need to be now though